src/main/java/edu/kis/powp/jobs2d/command/ICompoundCommand.java

```diff
@@ -2,11 +2,13 @@
 
 import java.util.Iterator;
 
+
 /**
  * Interface extending Job2dDriverCommand to execute more than one command.
  */
 public interface ICompoundCommand extends DriverCommand {
 
 	public Iterator<DriverCommand> iterator();
 
+	public ICompoundCommand clone() throws CloneNotSupportedException;
```

Comment:
It would be best to provide a default implementation. You could use the ICompoundCommandInstance.